### PR TITLE
[FIX] Membership search by name or email

### PR DESCRIPTION
**Pull request by antespi**
Fix for issue #6703
Official PR https://github.com/odoo/odoo/pull/6704
Adds a search field name for searching by partner display_name, name or email

### CODE REVIEW

File: addons/membership/membership_view.xml [hunk 1]
```diff
@@ -234,7 +235,7 @@
                                 <label for="membership_state"/>
                                 <div>
                                     <field name="membership_state"/>
-                                    <button name="%(action_membership_invoice_view)d" type="action" string="Buy Membership" 
+                                    <button name="%(action_membership_invoice_view)d" type="action" string="Buy Membership"
```
Comment (pedrobaeza) [inline addons/membership/membership_view.xml]: You are mixing here another issue

**Comment by pedrobaeza** [comment]
You should patch _name_search_ instead of overriding search view, because the problem will be there, but hidden. Using the search on a many2one will receive the same crash.

File: addons/membership/membership_view.xml (hunk 1 above)
Comment (antespi) [inline addons/membership/membership_view.xml]: This is an editor issue, it trims spaces at endlines. This is a good practice in programming.

Comment (pedrobaeza) [inline addons/membership/membership_view.xml]: Yeah, I know, but for keeping the diff clear and reduce possible upstream merge conflicts, we ignore those changes on OCB (not in the rest of the branches).

**Comment by antespi** [comment]
Sorry i don't understand your comment. This view renders res.partner model and membership inherits it but don't modify any method related with search or `_rec_name` class attribute.
How can I path name_search?

**Comment by dreispt** [comment]
I feel that searching Partners by email is a generally useful, and should something added directly to res.partner.

**Comment by pedrobaeza** [comment]
The error is usually caused by an error in the name_search method (in this case, of res.partner), that's why I say if you can patch that part.

**Comment by antespi** [comment]
name_search from res.partner is working in Sale and Purchase views.
Why i would have to change res.partner?

**Comment by pedrobaeza** [comment]
OK, let's other see if this is the best solution. I haven't deep enough into the problem to decide.

**Comment by gurneyalex** [comment]
I confirm I reproduce the crash on runbot.odoo.com and this branches fixes the issue. 

:+1:

**Comment by pedrobaeza** [comment]
@antespi, please restore the other line (with the final space), and I'll proceed with the merge, because I also confirmed that the fix works. :+1:

**Comment by yvaucher** [comment]
I put need fixing so we don't merge in this PR the extra unrelated change pointed by @pedrobaeza

**Comment by antespi** [comment]
@pedrobaeza i've just removed trailing space. I missed the notification

**Comment by pedrobaeza** [comment]
:+1:

**Pull request merged**